cmd/erasure-sets.go

```diff
@@ -357,9 +357,14 @@ func newErasureSets(ctx context.Context, endpoints Endpoints, storageDisks []Sto
 
 	mutex := newNSLock(globalIsDistErasure)
 
+	// Number of buffers, max 2GB.
+	n := setCount * setDriveCount
+	if n > 100 {
+		n = 100
+	}
 	// Initialize byte pool once for all sets, bpool size is set to
 	// setCount * setDriveCount with each memory upto blockSizeV1.
-	bp := bpool.NewBytePoolCap(setCount*setDriveCount, blockSizeV1, blockSizeV1*2)
+	bp := bpool.NewBytePoolCap(n, blockSizeV1, blockSizeV1*2)
```

Comment:
I suspect this may change performance since we will be trading off allocation time v/s re-using the buffer. Would it make sense to keep this sort of configurable? perhaps we can limit this with the value specified in the API request limit per node? 

```
bp := bpool.NewBytePoolCap(apiRequestMaxPerNode, blockSizeV1, blockSizeV1*2)
```
 
This would sufficiently restrict as per the total ram as well

Reply:
Actually come to think of it 100 is actually okay.